cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/RunRecordCorrectorService.java

```diff
@@ -146,6 +153,23 @@ void fixRunRecords() {
     return fixedPrograms;
   }
 
+  private ProgramOptions constructAndGetProgramOptions(RunRecordMeta record) {
```

Comment:
the run record corrector should not need to know how to create ProgramOptions from a run record. Also, why do we need to add things to the system properties instead of using the existing properties? Wouldn't this only apply to records that started before 5.0.0 but where corrected after 5.0.0?

Reply:
umm, are you saying system properties will include artifactId already and i am redoing a put?  i can verify if that is the case. please clarify.

Reply:
ProgramOptions do not change throughout the course of the run. They are only set once at the start. So if they are set in the beginning, there is no need to re-set them here. But like I mentioned elsewhere, I don't think we should be passing ProgramOptions at all, because they do not change.